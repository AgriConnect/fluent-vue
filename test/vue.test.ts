import { createLocalVue, shallowMount } from '@vue/test-utils'

import { FluentBundle, ftl } from 'fluent'

// @ts-ignore
import App from './components/App.vue'

import FluentVue from '../src/fluent-vue'

const bundle = new FluentBundle('en-US', {
  useIsolating: false
})

bundle.addMessages(ftl`
  message = Hello, { $name }!
`)

const fluent = new FluentVue({
  // locale: mainLocale,
  // fallbackLocale: getFallbackLocale(mainLocale),
  bundle
})

const localVue = createLocalVue()
localVue.use(FluentVue)

describe('vue integration', () => {
  it('works', () => {
    // Act
    const mounted = shallowMount(App, {
      localVue,
      fluent
    } as any)

    // Assert
    console.log(mounted)
  })
})
