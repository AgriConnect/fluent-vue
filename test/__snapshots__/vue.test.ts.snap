// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue integration works 1`] = `
<div>
  
  Hello, John!

</div>
`;
